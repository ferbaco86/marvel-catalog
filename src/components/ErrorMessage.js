import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const ErrorContainer = styled.div`
display: flex;
margin: 0 auto;`;

const ErrorText = styled.span`
color: ${props => props.textColor};
font-size: 2rem;
font-family: 'Roboto Condensed', sans-serif;
font-weight: 700;`;

const ErrorMessage = props => {
  const { message, color } = props;
  return (
    <ErrorContainer>
      <ErrorText textColor={color}>
        {' '}
        {message}
        {' '}
      </ErrorText>
    </ErrorContainer>
  );
};

ErrorMessage.defaultProps = { color: '#ec1d24' };
ErrorMessage.propTypes = {
  message: PropTypes.string.isRequired,
  color: PropTypes.string,
};

export default ErrorMessage;
