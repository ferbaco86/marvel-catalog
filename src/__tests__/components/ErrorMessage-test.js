import { render, screen, act } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import store from '../../reducers/index';
import ErrorMessage from '../../components/ErrorMessage';

describe('Rendering component', () => {
  it('creates a ErrorMessage component', () => {
    act(() => {
      render(
        <BrowserRouter>
          <Provider store={store}>
            <ErrorMessage message="Error!" />
          </Provider>
        </BrowserRouter>,
      );
    });
    screen.getByText('Error!');
  });
});

describe('Display', () => {
  it('renders correctly', () => {
    const snap = renderer.create(
      <BrowserRouter>
        <Provider store={store}>
          <ErrorMessage message="Error!" />
        </Provider>
      </BrowserRouter>,
    ).toJSON();
    expect(snap).toMatchSnapshot();
  });
});
