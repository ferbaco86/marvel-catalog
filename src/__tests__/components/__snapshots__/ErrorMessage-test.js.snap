// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Display renders correctly 1`] = `
<div
  className="sc-bdfBwQ dSDVZk"
>
  <span
    className="sc-gsTCUz eDDWpT"
  >
     
    Error!
     
  </span>
</div>
`;
